dpctl/sycl_core.pyx

```diff
@@ -250,6 +287,16 @@ cdef class SyclProgram:
         name = kernel_name.encode('utf8')
         return DPPLProgram_HasKernel(self._program_ref, name)
 
+    def addressof_ref (self):
+        """Returns the address of the C API DPPLSyclProgramRef pointer
+        as a long.
+
+        Returns:
+            The address of the DPPLSyclProgramRef object used to create this
+            SyclProgram cast to a long.
+        """
+        return int(<long>self._program_ref)
```

Comment:
@PokhodenkoSA @diptorupd 

Long is not good here. It is a 32-bit integer on Windows. We should be using `<size_t>` instead.